### PR TITLE
Modify S5496(Python): Migrate to LayC

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S5496/python/rule.adoc [hunk 1]
```diff
@@ -1,53 +1,120 @@
...
-User-provided data, such as URL parameters, POST data payloads, or cookies, should always be considered untrusted and tainted. Constructing HTML content directly from tainted data enables attacker(s) to inject special crafted values to abuse template engine's rendering process. Successful server side template injection (SSTI) attack can lead to arbitrary file read or operating system commands execution.
+Server-side template injections occur in an application when the application
+retrieves data from a user or a third-party service and inserts it into a
+template, without sanitizing it first.
+
+If an application contains a template that is vulnerable to injections, 
+it is exposed to attacks that target the underlying rendering server.
+
+A user with malicious intent carefully performs actions whose goal is to cause
+the template to change its logic into a malicious one.
```
Comment (daniel-teuchert-sonarsource) [inline rules/S5496/python/rule.adoc:12]: Maybe rephrase this sentence:
A user with malicious intent can create requests that will cause the template to change its logic into unwanted behavior.

File: rules/S5496/python/rule.adoc [hunk 2]
```diff
@@ -1,53 +1,120 @@
...
     username = request.args.get('username')
-    template = "<p>Hello {{ name }}</p>"
-    return render_template_string(template, name=username) # Compliant
+    template = "<p>Hello {{ username }}</p>"
+    return render_template_string(template, username=username)
 ----
 
+=== How does this work?
+
+== Use template variables
```
Comment (daniel-teuchert-sonarsource) [inline rules/S5496/python/rule.adoc:98]: Should this be a second- or a fourth-level heading?

**Review by daniel-teuchert-sonarsource — COMMENTED** [review]
Except for my two comments, everything looks good!

File: rules/S5496/python/rule.adoc (hunk 1 above)
Comment (loris-s-sonarsource) [reply rules/S5496/python/rule.adoc:12]: sgtm !

Comment (loris-s-sonarsource) [inline rules/S5496/python/rule.adoc:12]:
```suggestion
A user with malicious intent can create requests that will cause
the template to change its logic into unwanted behavior.
```

File: rules/S5496/python/rule.adoc (hunk 2 above)
Comment (loris-s-sonarsource) [reply rules/S5496/python/rule.adoc:98]: Oh you are right, I changed it mistakenly to a higher level header
```suggestion
==== Use template variables
```

**Review by daniel-teuchert-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**